hub/instruction_handling/txn/constraints.tex

```diff
@@ -0,0 +1,44 @@
+\begin{center}
+	\boxed{%
+		\text{The stack constraints presented below assume }
+		\begin{cases}
+			\peekStack       _{i} = 1           \\
+			\stackDecTxnFlag _{i} = 1           \\
+			\stackSux _{i} + \stackSox _{i} = 0 \\
+		\end{cases}}
+\end{center}
+We impose the following constraints:
+\begin{description}
+	\item[\underline{Setting the stack pattern:}] we impose $\zeroOneSP_{i}$;
+	\item[\underline{Setting $\nonStackRows$:}] we impose $\nonStackRows_{i} = 1 + \cmc_{i}$;
+	\item[\underline{Setting the peeking flags:}] we impose
+		\[
+			\left[ \begin{array}{lrcl}
+				+ & \peekTransaction _{i + \locTxnRow} \\
+				+ & \peekContext     _{i + \locConRow}  & \!\!\!\cdot\!\!\! & \cmc_{i} \\
+			\end{array} \right]
+			=
+			\nonStackRows_{i}
+		\]
+	\item[\underline{Setting the gas cost:}] we impose that $\gasCost_{i} = \decStaticGas_{i}$;
+	\item[\underline{Value constraints:}]
+		\If $\xAhoy _{i} = 0$ \Then we impose that
```

Comment:
This line here, making the result writing conditional to `XAHOY = 0`, is the only meaningful change.